Added Kubernetes Hardening security guide (#441)
gardener/documentation: website/documentation/guides/kubernetes-hardening/_index.md

---
title: Kubernetes Cluster Hardening Procedure for Cluster Admins
description: "Using Cloud Access Management (CAM) for compliant user management of your Gardener Projects"
level: advanced
category: Security
persona: Operators
publishdate: 2023-10-10
tags: ["task"]
---

## Overview

> The Gardener team takes security very seriously, which is why we mandate the Security Technical Implementation Guide (STIG) for Kubernetes as published by the Defense Information Systems Agency (DISA) [here](https://public.cyber.mil/stigs/downloads/). In combination with the control change above, this offers Gardener stakeholders the opportunity to adopt and show SCR03 compliance with DISA Kubernetes STIG as well. The latest release in machine readable format can be found in the [STIGs Document Library](https://public.cyber.mil/stigs/downloads/?_dl_facet_stigs=container-platform) by searching for Kubernetes.

Reviewer: @n-boshnakov IMHO we shall remove SCR03 as well.

---
title: Kubernetes Cluster Hardening Procedure for Cluster Admins
description: "Using Cloud Access Management (CAM) for compliant user management of your Gardener Projects"
level: advanced
category: Security
persona: Operators
publishdate: 2023-10-10
tags: ["task"]
---

## Overview

> The Gardener team takes security very seriously, which is why we mandate the Security Technical Implementation Guide (STIG) for Kubernetes as published by the Defense Information Systems Agency (DISA) [here](https://public.cyber.mil/stigs/downloads/). Thus, Gardener has decided and aligned with SGS to adopt and show SCR03 compliance with the DISA Kubernetes STIGs. In combination with the control change above, this offers Gardener stakeholders the opportunity to adopt and show SCR03 compliance with DISA Kubernetes STIG as well. The latest release in machine readable format can be found in the [STIGs Document Library](https://public.cyber.mil/stigs/downloads/?_dl_facet_stigs=container-platform) by searching for Kubernetes.

Reviewer: > Thus, Gardener has decided and aligned with SGS to adopt and show SCR03 compliance with the DISA Kubernetes STIGs.

This sentence should be deleted. We shall not mention SGS in the public site (at least not as of today).
We shall rethink the whole paragraph.

Reviewer: On top of that, probably as additional public page, we shall list all rules, which are responsibility to Service Provider (this can be a 2nd step)

|242442|Kubernetes must remove old components after updated versions have been installed|:x:| While Gardener manages all its components in its system namespaces (automated), you are naturally responsible for your own workload.|
|254800|Kubernetes must have a Pod Security Admission control file configured|:x:| Gardener ensures that the pod security configuration allows system components to be deployed in the kube-system namespace but does not set configurations that can affect user namespaces. It is recommended that users enforce a minimum of [baseline pod security level for their workload](https://kubernetes.io/docs/concepts/security/pod-security-admission/#pod-security-levels) via [PodSecurity admission plugin](https://gardener.cloud/docs/gardener/pod-security/#admission-configuration-for-the-podsecurity-admission-plugin). |

> The Gardener infrastructure is taking care of the rules which are part of DISA Kubernetes STIGs but not listed here (secure-by-design/code).

Reviewer: > The Gardener infrastructure is taking care of the rules which are part of DISA Kubernetes STIGs but not listed here (secure-by-design/code).

This sentence is not guaranteed to be true outside SAP, is it?

---

## Overview

> The Gardener team takes security very seriously, which is why we mandate the Security Technical Implementation Guide (STIG) for Kubernetes as published by the Defense Information Systems Agency (DISA) [here](https://public.cyber.mil/stigs/downloads/). In combination with the control change above, this offers Gardener adopters the opportunity to show compliance with DISA Kubernetes STIG. The latest release in machine readable format can be found in the [STIGs Document Library](https://public.cyber.mil/stigs/downloads/?_dl_facet_stigs=container-platform) by searching for Kubernetes.

Reviewer: ```suggestion
The Gardener team takes security seriously, which is why we mandate the Security Technical Implementation Guide (STIG) for Kubernetes as published by the Defense Information Systems Agency (DISA) [here](https://public.cyber.mil/stigs/downloads/). We offer Gardener adopters the opportunity to show compliance with DISA Kubernetes STIG via the compliance checker tool [diki](https://github.com/gardener/diki). The latest release in machine readable format can be found in the [STIGs Document Library](https://public.cyber.mil/stigs/downloads/?_dl_facet_stigs=container-platform) by searching for Kubernetes.
```
